### ISSUE TITLE
Support expecting a variable to be equal to an AnyClass (AnyObject.Type) type 

### ISSUE THREAD

**Issue by glentregoning** [Quick/Nimble#252]
AnyClass the type alias to AnyObject.Type doesn't implement the Equatable protocol.

As a result the following comparison doesn't work.

```
expect(object.dynamicType).to(equal(MyClass.self))
```

The following is possible, but isn't ideal given it isn't consistent with the common matcher style syntax. 

```
expect(object.dynamicType is MyClass.self).to(beTrue())
```

**Comment by ikesyo** [comment]
I think `expect(object.dynamicType is MyClass.self).to(beTrue())` would be fine.

**Comment by glentregoning** [comment]
@ikesyo that indeed works, but not ideal.

Perhaps I was a little strong saying this is impossible. I've updated my answer to say this isn't possible using the traditional matcher syntax.

**Comment by jeffh** [comment]
I think this would be great to have.

**Comment by ikesyo** [comment]
Resolved by #366.
